web/src/engine/websites/MultPorn.ts

```diff
@@ -0,0 +1,107 @@
+@Common.PagesSinglePageJS(pageScript, 1500)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = `${this.URI.origin}/views/ajax`;
+
+    public constructor() {
+        super('multporn', 'MultPorn', 'https://multporn.net', Tags.Media.Comic, Tags.Media.Manga, Tags.Language.Multilingual, Tags.Source.Aggregator, Tags.Rating.Pornographic);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let category of categories) {
+            const mangas = await Common.FetchMangasMultiPageCSS.call(this, provider, category + '?page={page}', 'div.view-content table tr td strong a', 0);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const url = new URL(manga.Identifier, this.URI);
+        const settings = await FetchWindowScript<DrupalSettings>(new Request(url), 'Drupal.settings');
+        const chapterList = [];
+
+        if (settings.views?.ajaxViews) {
+            for (let page = 0, run = true; run; page++) {
+                const chapters = await this.GetChaptersFromAjaxPage(manga, page, settings);
+                chapters.length > 0 ? chapterList.push(...chapters) : run = false;
+            }
+            return chapterList;
+        } else {
+            return Common.FetchChaptersSinglePageCSS.call(this, manga, 'div.view-content table tr td strong.field-content a');
+        }
+
+    }
+
+    private async GetChaptersFromAjaxPage(manga: Manga, page: number, settings: DrupalSettings): Promise<Chapter[]> {
+
+        const view = Object.values(settings.views.ajaxViews).shift();
+        const params = new URLSearchParams();
+        for (const key of Object.keys(view)) {
+            params.append(key, view[key]);
+        };
+        params.set('page', `0,${page}`);
+
+        const results = await FetchJSON<DrupalResult[]>(new Request(this.apiUrl, {
+            method: 'POST',
+            body: params.toString(),
+            headers: {
+                Accept: 'application/json, text/javascript, */*; q=0.01',
+                'Content-type': 'application/x-www-form-urlencoded; charset=UTF-8',
+                'X-Requested-With': 'XMLHttpRequest',
+                Origin: this.URI.origin,
+                Referer: new URL(manga.Identifier, this.URI).href
+            }
+        }));
+
+        const { data } = results.find(entry => entry.command == 'insert');
+        const dom = new DOMParser().parseFromString(data, 'text/html');
+        const nodes = dom.querySelectorAll<HTMLAnchorElement>('div.view-content table tr td strong.field-content a');
+        return [...nodes]?.map(chapter => new Chapter(this, manga, chapter.pathname, chapter.text.trim())) ?? [];
```

Comment:
Null coalescing operator `[...]?.` is redundant, an empty array can never be `null`